pkg/translate/translate.go

```diff
@@ -109,10 +109,22 @@ type Translation struct {
 
 // NewTranslator creates a new Translator for minorVersion and returns a ptr to it.
 func NewTranslator(minorVersion version.MinorVersion) (*Translator, error) {
+	return newTranslator(minorVersion, 1)
+}
+
+func newTranslator(minorVersion version.MinorVersion, fallbackNum uint) (*Translator, error) {
 	v := fmt.Sprintf("%s.%d", minorVersion.MajorVersion, minorVersion.Minor)
 	f := "translateConfig/translateConfig-" + v + ".yaml"
 	b, err := vfs.ReadFile(f)
 	if err != nil {
+		if fallbackNum > 0 && minorVersion.Minor > 0 {
+			major := minorVersion.Major
+			minor := minorVersion.Minor - 1
+			previous := version.NewMinorVersion(major, minor)
+			log.Warnf("could not read translateConfig file for version %s, fallback to version %s",
+				minorVersion, previous)
+			return newTranslator(previous, fallbackNum-1)
```

Comment:
I don't quite get the usage of fallback num here, so looks like it is hardcoded to 1 at the exported method, then in this recursion, next call would always have fallbacknum equals 0 so it has only two states. If that's the case why we still need to call newTranslator function again

Reply:
I can rename the '1' to a constant `maxFallbackNum`. Sorry, I do not get your idea. What's your suggestion?

Reply:
Fixed.